gradle-plugin/src/main/kotlin/com/github/buildeye/inputs/snapshot/InputsSnapshot.kt

```diff
@@ -0,0 +1,8 @@
+package com.github.buildeye.inputs.snapshot
+
+import java.io.Serializable
+
+data class InputsSnapshot(
+        val properties: Map<String, Any?>,
+        val inputFiles: Map<String, String>
+) : Serializable
```

Comment:
I recommend to set `serialVersionUID` explicitly for all `Serializable` classes to avoid unexpected `InvalidClassException` (e.g. when different JVMs are user for serialization/deserialization): 
```
data class InputsSnapshot(
        val properties: Map<String, Any?>,
        val inputFiles: Map<String, String>
) : Serializable {
    companion object {
        private const val serialVersionUID = 0L
    }
} 
```
See also https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html